helix-term/src/ui/picker.rs

```diff
@@ -41,6 +41,7 @@ use helix_view::{
 use super::{menu::Item, overlay::Overlay};
 
 pub const MIN_AREA_WIDTH_FOR_PREVIEW: u16 = 72;
+pub const MIN_AREA_HEIGHT_FOR_PREVIEW: u16 = 18;
```

Comment:
I think the a simple cutoff like this isn't quite ideal. For example:

This looks very readable:

![image](https://github.com/helix-editor/helix/assets/61850714/5e093b23-cd99-45cc-b01d-fcb198ee1821)

but making the picker slightly wider switches to a critical split which feels worse to me:

![image](https://github.com/helix-editor/helix/assets/61850714/4aae0116-e456-47fe-b8e3-24c12cff5da6)


I think it might make more sense to have 2 different thresholds here. For wide windows (lets say more than 120 wide)  we always show the current layout, if that's not the case and we are higher that a fairly large height (lets say 40) always shows the vertical layout and if that's not true you try the current fallback. I think we would get slightly nicer layouts that way (there could also be other implementation that work better but its what came to mind)